app/src/composables/use-mutex.ts

```diff
@@ -0,0 +1,59 @@
+import { sleep } from '@directus/sdk';
+
+const MutexKey = ['auth_refresh'] as const;
+
+type MutexKey = (typeof MutexKey)[number];
+
+const timeout = 500;
+const maxRetries = 10;
+
+export function useMutex(key: MutexKey, expiresMs: number) {
+	const internalKey = `directus-mutex-${key}`;
+	const useWebLock = !!navigator.locks;
+
+	async function acquireMutex(callback: (lock?: Lock | null) => Promise<any>): Promise<any> {
```

Comment:
I think we need a timeout if the lock should not be released for some reason, either here in `useMutex` (use `expiresMs`?) or on caller side

https://developer.mozilla.org/en-US/docs/Web/API/LockManager/request#signal_example